src/cubeb_audiounit.cpp

```
@@ -741,33 +741,22 @@ static int audiounit_stream_get_volume(cubeb_stream * stm, float * volume);
 static int audiounit_stream_set_volume(cubeb_stream * stm, float volume);
 
 static int
-audiounit_set_device_info(cubeb_stream * stm, AudioDeviceID id, io_side side)
+audiounit_set_device_info(cubeb_stream * stm, AudioDeviceID id, cubeb_device_type type)
 {
   assert(stm);
+  assert(type == CUBEB_DEVICE_TYPE_INPUT || type == CUBEB_DEVICE_TYPE_OUTPUT);
 
-  device_info * info = nullptr;
-  cubeb_device_type type = CUBEB_DEVICE_TYPE_UNKNOWN;
+  device_info * info = (type == CUBEB_DEVICE_TYPE_INPUT)
+    ? &stm->input_device
+    : &stm->output_device;
 
-  if (side == io_side::INPUT) {
-    info = &stm->input_device;
-    type = CUBEB_DEVICE_TYPE_INPUT;
-  } else if (side == io_side::OUTPUT) {
-    info = &stm->output_device;
-    type = CUBEB_DEVICE_TYPE_OUTPUT;
-  }
   memset(info, 0, sizeof(device_info));
   info->id = id;
-
-  if (side == io_side::INPUT) {
-    info->flags |= DEV_INPUT;
-  } else if (side == io_side::OUTPUT) {
-    info->flags |= DEV_OUTPUT;
-  }
+  info->flags |= (type == CUBEB_DEVICE_TYPE_INPUT) ? DEV_INPUT : DEV_OUTPUT;
 
   AudioDeviceID default_device_id = audiounit_get_default_device_id(type);
-  if (default_device_id == kAudioObjectUnknown) {
-    return CUBEB_ERROR;
-  }
+  assert(default_device_id != kAudioObjectUnknown);
```

This will crash when there is no device. See #358 . We need to return an error.

---

You're right. Thanks!